### PR TITLE
Move echobot into /var/lib/echobot

### LINKED ISSUE #256: Echobot is running as root in root
Echobot service currently starts echobot as root and without working directory, so it sets up its account directory in `/accounts`.

We should run it as separate user with working directory in `/home/echobot`.

The problem with redeploying echobot is that it generates new password on deployment and takes the name first time it starts. I suggest that instead of allowing anyone to take "echo" username in
https://github.com/deltachat/chatmail/blob/c640087498f891dd51d74dac9912f9fbd5d8a7b3/chatmaild/src/chatmaild/doveauth.py#L53
we place "echo" password into chatmail.ini and then both bot and doveauth take it from there.

### LINKED ISSUE #277: Creating echo@staging.testrun.org doesn't work: Authentication Failed.
- Operating System (Linux/Mac/Windows/iOS/Android): Debian 12
- Delta Chat Version: Core v1.137.4
- Expected behavior: If you write to echo@staging.testrun.org, you get your message back as a reply.
- Actual behavior: echo@staging.testrun.org doesn't reply, because the account fails to get created.
- Steps to reproduce the problem:
  - deploy staging.testrun.org
  - send a message to echo@staging.testrun.org
  - see in the logs that the echobot service failed: `journalctl -u echobot.service`
- Logs:

```
Mai 02 11:19:14 staging echobot[13517]: [2024-05-02T11:19:14Z INFO  deltachat_rpc_server] Starting with accounts directory `accounts`.
Mai 02 11:19:14 staging echobot[13517]: [2024-05-02T11:19:14Z INFO  deltachat_rpc_server] Creating JSON-RPC API.
Mai 02 11:19:14 staging echobot[13515]: INFO:root:Running deltachat core v1.137.4
Mai 02 11:19:23 staging echobot[13517]: [2024-05-02T11:19:23Z INFO  deltachat_rpc_server] EOF reached on stdin
Mai 02 11:19:23 staging echobot[13515]: Traceback (most recent call last):
Mai 02 11:19:23 staging echobot[13515]:   File "/usr/local/lib/chatmaild/venv/bin/echobot", line 8, in <module>
Mai 02 11:19:23 staging echobot[13515]:     sys.exit(main())
Mai 02 11:19:23 staging echobot[13515]:              ^^^^^^
Mai 02 11:19:23 staging echobot[13515]:   File "/usr/local/lib/chatmaild/venv/lib/python3.11/site-packages/chatmaild/echo.py", line 82, in main
Mai 02 11:19:23 staging echobot[13515]:     bot.configure(email, password)
Mai 02 11:19:23 staging echobot[13515]:   File "/usr/local/lib/chatmaild/venv/lib/python3.11/site-packages/deltachat_rpc_client/client.py", line 202, in configure
Mai 02 11:19:23 staging echobot[13515]:     super().configure(email, password, **kwargs)
Mai 02 11:19:23 staging echobot[13515]:   File "/usr/local/lib/chatmaild/venv/lib/python3.11/site-packages/deltachat_rpc_client/client.py", line 87, in configure
Mai 02 11:19:23 staging echobot[13515]:     self.account.configure()
Mai 02 11:19:23 staging echobot[13515]:   File "/usr/local/lib/chatmaild/venv/lib/python3.11/site-packages/deltachat_rpc_client/_utils.py", line 201, in wrapper
Mai 02 11:19:23 staging echobot[13515]:     return f()
Mai 02 11:19:23 staging echobot[13515]:            ^^^
Mai 02 11:19:23 staging echobot[13515]:   File "/usr/local/lib/chatmaild/venv/lib/python3.11/site-packages/deltachat_rpc_client/_utils.py", line 193, in f
Mai 02 11:19:23 staging echobot[13515]:     generator.send(res())
Mai 02 11:19:23 staging echobot[13515]:                    ^^^^^
Mai 02 11:19:23 staging echobot[13515]:   File "/usr/local/lib/chatmaild/venv/lib/python3.11/site-packages/deltachat_rpc_client/rpc.py", line 28, in __call__
Mai 02 11:19:23 staging echobot[13515]:     raise JsonRpcError(response["error"])
Mai 02 11:19:23 staging echobot[13515]: deltachat_rpc_client.rpc.JsonRpcError: {'code': -1, 'message': 'Cannot login as "echo@staging.testrun.org". Please check if the email address and the password are correct.\n\nno response: code: None, info: Some("[AUTHENTICATIONFAILED] Authentication failed.")'}
Mai 02 11:19:23 staging systemd[1]: echobot.service: Main process exited, code=exited, status=1/FAILURE
Mai 02 11:19:23 staging systemd[1]: echobot.service: Failed with result 'exit-code'.
```

### PR DESCRIPTION
**Pull request by link2xt**
Fixes #256 
Fixes #277

### CODE REVIEW

**Review by hpk42 — COMMENTED** [review]
I think it's better to generate and store the echbot password in its home dir.  The ini file better not contains secrets.

File: chatmaild/src/chatmaild/doveauth.py [hunk 1]
```diff
@@ -45,23 +46,30 @@ def is_allowed_to_create(config: Config, user, cleartext_password) -> bool:
...
+            config.username_max_length,
+        )
 
     return True
 
 
 def get_user_data(db, config: Config, user):
+    if user == f"echo@{config.mail_domain}":
+        return dict(
+            home=f"/home/vmail/mail/{config.mail_domain}/echo", uid="vmail", gid="vmail"
```
Comment (missytake) [inline chatmaild/src/chatmaild/doveauth.py:70]:
```suggestion
            home=f"/home/vmail/mail/{config.mail_domain}/echo@{config.mail_domain}", uid="vmail", gid="vmail"
```

Typically, the vmail directory would be `/echo@nine.testrun.org,` not `/echo` I think...

**Review by missytake — APPROVED** [review]
Looks good :)

**Review by hpk42 — COMMENTED** [review]
nice

**Pull request merged**